packages/web-components/src/components/fade-in-out/fade-in-out.ts

```diff
@@ -0,0 +1,256 @@
+ *
+ * As an example, the function can be called to target all instances of the
+ * elements in a list:
+ *
+ * const list = 'bx--content-block, bx--content-group';
+ *
+ * For default values of 400ms and 'one and done' play:
+ * <dds-fade-in-out selectorTargets="${selectorTargets}"/>
+ *
+ * With 'continuous play' option:
+ * <dds-fade-in-out selectorTargets="${selectorTargets}" iterations="true"/>
+ *
+ * For custom delay time, set within targeted class in the application's CSS code as such:
+ *
+ * .bx--content-block {
+ *   --#{$dds-prefix}--fade-in-out-delay: 250ms;
+ * }
+ *
+ * @element dds-fade-in-out
+ */
+@customElement(`${ddsPrefix}-fade-in-out`)
+class DDSFadeInOut extends StableSelectorMixin(LitElement) {
+  /**
+   * The CSS selector that selects all fade in-out target elements.
+   */
+  @property({ attribute: 'selectorTargets' })
+  selectorTargets?: string;
+
+  /**
+   * Iteration boolean for continuous play option.
+   *
+   * `true` to keep fade in/out behavior during the entire component lifecycle,
+   * `false` to do it only once.
+   *
+   */
+  @property({ type: Boolean, attribute: 'keep-animation' })
+  keepAnimation = false;
+
+  /**
+   * Amount of columns used for calculation.
+   *
+   * @private
+   */
+
+  private _colSpan = 3;
+
+  /**
+   * Saved list of elements to observe to avoid calling querySelectorAll
+   * more than once.
+   *
+   * @private
+   */
+  private _elements: HTMLElement[] = [];
+
+  /**
+   * Intersection Observer that watches outer viewport.
+   *
+   * @private
+   */
+  private _rootObserver: IntersectionObserver | null = null;
+
+  /**
+   * Intersection observer that watches the inner viewport.
+   *
+   * @private
+   */
+  private _innerObserver: IntersectionObserver | null = null;
+
+  /**
+   * Resize observer to trigger rootMargin recalculations
+   *
+   * @private
+   */
+  private _resizeObserver: any | null = null;
+
+  /**
+   * Intersection Observer options
+   *
+   * @private
+   */
+  private _options = {
+    rootMargin: '0px',
+    threshold: 0,
+  };
+
+  /**
+   * The inner viewport calculation for the root margins.
+   *
+   * @private
+   */
+  private _getViewportMargin() {
+    return `-${((document.documentElement.clientHeight * this._colSpan) / breakpoints.max.columns).toString()}px 0px`;
+  }
+
+  /**
+   * Cleans observers upon update or exit, and creates new instances if needed.
+   *
+   * @param {boolean} create to create new observer instances
+   */
+  private _cleanAndCreateObservers({ create }: { create?: boolean } = {}) {
+    this._cleanAndCreateRootObserver();
+    this._cleanAndCreateInnerObserver();
+    this._cleanAndCreateResizeObserver();
+
+    if (create) {
+      this._cleanAndCreateRootObserver({ create: true });
+      this._cleanAndCreateResizeObserver({ create: true });
+
+      const { selectorTargets } = this;
+      if (selectorTargets) {
+        forEach(document.querySelectorAll(selectorTargets), item => {
```

Comment:
The `qSA` should be scoped in the host element of this custom element:

```suggestion
        forEach(this.querySelectorAll(selectorTargets), item => {
```